### DISCUSSION TITLE
Изображение на сайт. Поздравление с праздником победы 9 мая

### DISCUSSION THREAD

**Discussion by audetv** [terratensor/svodd#177] [Идеи]
Товарищи, появилась идея разместить на сайте к 9 мая изображение - поздравление с праздником победы.
Например разместить картинку над строкой поиска ( форма прямоугольника по ширине строки и с высотой меньше, чем ширина, в форме баннера).
Напишите поддерживаете ли идею и пишите в комментарии, что должно быть размещено на изображении.
Например:

Фон изображения красный. Текст: Поздравляем с Днем Победы!

1. На фоне фрагменты праздничного салюта, изображен [Мавзолей Ленина](https://yandex.ru/search/?text=%D0%BC%D0%B0%D0%B2%D0%B7%D0%BE%D0%BB%D0%B5%D0%B9+%D0%BB%D0%B5%D0%BD%D0%B8%D0%BD%D0%B0&lr=2&src=suggest_B), к подножью которого бросают фашистские знамена. 
3. [Родина-Мать.](https://yandex.ru/search/?text=%D0%A0%D0%BE%D0%B4%D0%B8%D0%BD%D0%B0-%D0%9C%D0%B0%D1%82%D1%8C.&lr=2)
4. [Воин-освободитель.](https://yandex.ru/search/?text=%D0%92%D0%BE%D0%B8%D0%BD-%D0%BE%D1%81%D0%B2%D0%BE%D0%B1%D0%BE%D0%B4%D0%B8%D1%82%D0%B5%D0%BB%D1%8C.&lr=2)
5. [Орден «Победа»](https://yandex.ru/search/?text=%D0%BE%D1%80%D0%B4%D0%B5%D0%BD+%C2%AB%D0%BF%D0%BE%D0%B1%D0%B5%D0%B4%D0%B0%C2%BB%2C+%D1%83%D1%87%D1%80%D0%B5%D0%B6%D0%B4%D0%B5%D0%BD%D0%BD%D1%8B%D0%B9+%D1%83%D0%BA%D0%B0%D0%B7%D0%BE%D0%BC+%D0%BF%D1%80%D0%B5%D0%B7%D0%B8%D0%B4%D0%B8%D1%83%D0%BC%D0%B0+%D0%B2%D0%B5%D1%80%D1%85%D0%BE%D0%B2%D0%BD%D0%BE%D0%B3%D0%BE+%D1%81%D0%BE%D0%B2%D0%B5%D1%82%D0%B0+%D1%81%D1%81%D1%81%D1%80+%D0%BE%D1%82+8+%D0%BD%D0%BE%D1%8F%D0%B1%D1%80%D1%8F+1943+%D0%B3.&lr=2&src=suggest_Pers), учрежденный Указом Президиума Верховного Совета СССР от 8 ноября 1943 г.
6. Орден СССР, учреждённый указом Президиума Верховного Совета СССР «[Об учреждении Ордена Отечественной войны I и II степени](https://yandex.ru/search/?text=+%D0%9E%D1%80%D0%B4%D0%B5%D0%BD+%D0%9E%D1%82%D0%B5%D1%87%D0%B5%D1%81%D1%82%D0%B2%D0%B5%D0%BD%D0%BD%D0%BE%D0%B9+%D0%B2%D0%BE%D0%B9%D0%BD%D1%8B&lr=2)» от 20 мая 1942 года

**Comment by iprst** [comment]
[Самый просматриваемый момент](https://youtu.be/7Xm5U2KHUaE?t=591) документальной хроники о Параде Победы 1945 года на Красной Площади.

**Reply by audetv** [reply]
Может вырезать этот фрагмент и транслировать его под строкой поиска на главной странице?
![2023-05-02_17-42-05](https://user-images.githubusercontent.com/10896447/235702776-00c39e9b-d18f-4898-8779-4e70410eaf37.jpg)

**Reply by audetv** [reply]
Или вообще весь парад целиком.

**Reply by iprst** [reply]
Идея с фрагментом мне нравится. Кто что скажет ещё?

**Reply by grigorovsckaya-sandra** [reply]
> Идея с фрагментом мне нравится. Кто что скажет ещё?

не совсем представляю, как это будет выглядеть, но я "ЗА" идею отметить 9 мая, и не только в течение одного дня, а фрагмент надо бы выбрать с более ясным и отчётливым видом брошенных фашистских штандартов, там есть более удачные кадры (нпр., 10.06 - 10.09, 10.26)

**Reply by iprst** [reply]
Я сделал видео в формате 1920×480 из трёх источников, слева ч/б хроника Гостелерадиофонда, справа цветная хроника ЦСДФ, а посередине плашка с красным колышущимся знаменем, Орденом Победы и надписью. Только надпись нужно придумать. Думал что-то вроде «Деды победили и мы победим!» и «С Днём Победы! 9 мая 1945 года». 

Скрин из АЕ:

![image](https://user-images.githubusercontent.com/129882753/235774579-3f49ba03-a8b0-4b45-8ab9-91d703b4aa53.png)

**Reply by grigorovsckaya-sandra** [reply]
может, "Победа всегда с нами!"?

**Reply by iprst** [reply]
> может, "Победа всегда с нами!"?

Думаю нужно что-то классическое. Пока сделал чередование «С Днём Победы! 9 мая 1945 года» и «Деды победили — и мы победим!» по 12-15 секунд в течение ролика длиной около минуты.

**Comment by iprst** [comment]
Загрузил видео, 50 мб. [Ссылка](https://drive.google.com/file/d/1WInFLRAOXSYOp_aaqE0MXn39Gn63XByK/view?usp=sharing).

**Reply by iprst** [reply]
Подразумевается, что видео показывается в резиновом блоке на всю ширину браузера. Если ширина странцы более 1/2 формата кадра, то есть более 960 пикселей, то кадр показывается целиком. Если ширина страницы меньше, то видео центрируется и показывается в натуральном размере с обрезанием кадра справа и слева. Возможны другие варианты.

**Reply by audetv** [reply]
Отлично, добавлю на сайт утром/днём 03.05.2023. Протестируем и решим когда запускать.

**Comment by audetv** [comment]
Добавил в режиме тестирования 2 варианта
А - вариант. попытался сделать так:

> Подразумевается, что видео показывается в резиновом блоке на всю ширину браузера. Если ширина странцы более 1/2 формата кадра, то есть более 960 пикселей, то кадр показывается целиком. Если ширина страницы меньше, то видео центрируется и показывается в натуральном размере с обрезанием кадра справа и слева. Возможны другие варианты.

B - вариант. Видео размещено в контейнере и уменьшается полностью пропорционально с размером экрана

https://svodd.ru/?feature=09051945A
https://svodd.ru/?feature=09051945B

Сейчас работает только если к адресу добавить параметр(feature=09051945A/B) или просто переходите по ссылкам выше.
Мысли, какой вариант размещаем?

**Reply by iprst** [reply]
В первом варианте при растягивании окна на всю ширину видео обрезается по высоте, так не должно быть.

**Reply by iprst** [reply]
Проще всего сделать резиновый контейнер на всю ширину страницы, поведение такое же как во втором варианте, но ширина на весь сайт.

**Reply by audetv** [reply]
> В первом варианте при растягивании окна на всю ширину видео обрезается по высоте, так не должно быть.

Обрезается? Не совсем понимаю, высота 480pх выставлена. Может из-за object-fit=contain в контейнере.

Но это и ладно, если делать второй вариант на всю ширину, без условия что видео центруется и обрезается по бокам если экран уменьшается, то да проще. Сделал обновил.

B:
https://svodd.ru/?feature=09051945B
В результатах поиска показываем?
https://svodd.ru/?search%5Bquery%5D=%D0%BF%D0%BE%D0%B1%D0%B5%D0%B4%D0%B0&search%5Bmatching%5D=&search%5Bmatching%5D=query_string&search%5Bdate%5D=&search%5Bdictionary%5D=0&feature=09051945B


В первом варианте убрал border 2px, он точно обрезал высоту
A:
https://svodd.ru/?feature=09051945A
В результатах поиска
https://svodd.ru/?search%5Bquery%5D=%D0%BF%D0%BE%D0%B1%D0%B5%D0%B4%D0%B0&search%5Bmatching%5D=&search%5Bmatching%5D=query_string&search%5Bdate%5D=&search%5Bdictionary%5D=0&feature=09051945A

**Reply by audetv** [reply]
Мне вариант B нравится больше.

**Reply by iprst** [reply]
> Может из-за object-fit=contain в контейнере.

Да, это что-то с вёрсткой контейнера. 

> если делать второй вариант на всю ширину

Этот вариант работает лучше всего, но тоже с контейнером какая-то проблема — справа на сайт добавляется небольшой отступ, прямо для всего <body>, и только блок <header class navbar navbar…> отрисовывается поверх этого отступа:

![chrome_VOoChB0T6h](https://user-images.githubusercontent.com/129882753/235909299-d0e2ebc5-285e-49a0-b214-ccfccf94a48e.gif)

**Reply by iprst** [reply]
> В результатах поиска показываем?

Думаю можно недельку показать. А может и не недельку.

**Reply by audetv** [reply]
понял, да. смотрю что это, сейчас поправлю

**Reply by iprst** [reply]
Но в целом нужно оставлять вариант B на весь экран, выглядит отлично, если убрать этот отступ и тем самым избавиться от появления горизонтальной прокрутки внизу страницы.

**Reply by audetv** [reply]
Посмотрел на мобильном, не всегда сразу видео загружается, появляется через некоторое время. Можно добавить в свойство тэга video poster="video/denpobedy.jpg"  по идее это изображение должно будет показываться до того как запуститься видео. Т.е не просто черное пустое место.

**Reply by iprst** [reply]
> Можно добавить в свойство тэга video poster="video/denpobedy.jpg"

Вот:

![image](https://user-images.githubusercontent.com/129882753/235912299-0c5cca31-ae05-4416-9e82-6413820d114e.png)

**Reply by audetv** [reply]
Обновил. Оставляем так? И уже запустил без параметров в url или к выходным ближе запустить? Я думаю, пусть уже показывает.
https://svodd.ru/

upd С постером на мобильном как и ожидал работает. Пока видедо грузится показывает изображение.

**Reply by iprst** [reply]
Да, оставляем, отлично всё.

**Comment by Redd-devel** [comment]
Может имеет смысл развить тему со значимыми для нашей Родины датами на стартовой странице поисковика? Для примера: день Великой Октябрьской Социалистической революции или день Космонавтики. За 1-1,5 недели помещать на стартовую страницу баннер или видео соответствующие. Если идея будет поддержана, то можно на год вперед отобрать события, подготовить для них материалы и менять их автоматически.

**Reply by iprst** [reply]
Не проблема. Нужен календарь значимых событий и предложения по их оформлению.

**Reply by audetv** [reply]
Поддерживаю. Да, надо составить календарь. Можно сделать отдельный issue и там вести работу.
Ещё идея, можно предыдущие поздравления оформлять в качестве архива, например, давать ссылку или короткую ссылку на хронологию в поиске, с датами 09.05.2023 - 10.05.2023 или другое окно 08.05.2023 - 10.05.2023 . И чтобы можно было по этой ссылке увидеть комментарии или иные события, когда подключим их в поиск, почитать поздравления, посмотреть, поздравительные видео или изображения, которые были сделаны сообществом к праздничной дате. 

★ https://svodd.ru/ZA8F5CIN 9 мая 2023 г. День Победы!

**Reply by Redd-devel** [reply]
По календарю наработки есть у Александры @grigorovsckaya-sandra и можно их использовать. По оформлению: верстка, созданная под 9 мая годится как шаблон, в котором может содержаться разный контент (видео, gif-файлы, css-анимация, аудио, статичные картинки)? Если да, то можно начать, как минимум обсуждение, того, чем наполнить значимое событие. Мне оформление стартовой страницы нравится (за что, кстати, огромное спасибо!).
Сами события уже есть на ФКТ, поэтому можно начинать их "парсинг".
Имеет ли смысл создать под календарь новое обсуждение?

**Reply by Redd-devel** [reply]
> Поддерживаю. Да, надо составить календарь. Можно сделать отдельный issue и там вести работу. Ещё идея, можно предыдущие поздравления оформлять в качестве архива, например, давать ссылку или короткую ссылку на хронологию в поиске, с датами 09.05.2023 - 10.05.2023 или другое окно 08.05.2023 - 10.05.2023 . И чтобы можно было по этой ссылке увидеть комментарии или иные события, когда подключим их в поиск, почитать поздравления, посмотреть, поздравительные видео или изображения, которые были сделаны сообществом к праздничной дате.
> 
> ★ https://svodd.ru/ZA8F5CIN 9 мая 2023 г. День Победы!

Алексей, спасибо! Пока я писал свой пост вы прочитали мои мысли про отдельный issue. Соборность!

**Reply by iprst** [reply]
> По календарю наработки есть у Александры @grigorovsckaya-sandra и можно их использовать.

С этого можно начать. 

> Если да, то можно начать, как минимум обсуждение, того, чем наполнить значимое событие.

Можно посмотреть каков подход к описанию сущностей на гуглобусе. Это связанные элементы: события — это один из типов сущностей геоматрицы. Приглашение в приватный репозиторий я отправил. Повторюсь, чтобы не было недосказанного — события, о которых идёт речь 100% являются объектом интереса геоматрицы и будут там представлены. Можно прямо оттуда и начинать, поскольку там уже есть десятки тысяч событий. Не всегда речь о тех событиях, которые подразумеваются в этом обсуждении, но это лишь различие в классе, типизации отдельных событий. Очевидно, что праздничные или трагичные российские события представляют собой интерес для всех. 

> По оформлению: верстка, созданная под 9 мая годится как шаблон, в котором может содержаться разный контент (видео, gif-файлы, css-анимация, аудио, статичные картинки)?

Принципиально можно использовать любой тип медиа. Который, разумеется, лезет в html. Сделать можно что угодно, будь вектор целей и предложенная ПФУ. Нужно только сразу различить — размещение записей в базе (например событий) и получение их из базы для публикации в каком-то оформлении, это разные процессы, которые можно (иногда нужно) разделить друг от друга сколь угодно далеко. 

> Имеет ли смысл создать под календарь новое обсуждение?

Есть чёткое предположение, что календарь потребует не просто обсуждения, а отдельного репозитория. Поскольку нет смысла делать одну работу дважды — календарь / события уже есть в геоматрице и существуют в виде комментариев.

**Comment by audetv** [comment]
Оставляем ещё видео поздравление на главной и в поиске?

**Reply by iprst** [reply]
Лично мне не мешает пользоваться поиском. Можно позже сделать какие-то другие видео или банеры и прочая, но нужны цели-задачи. Если убрать, тоже не критично, можно и убрать, но я бы оставил и пусть висит дальше, пока не придумали как дальше должно быть.

**Reply by audetv** [reply]
Мне тоже не мешает, значит пока пусть висит.
